packages/scene-composer/src/components/panels/scene-settings/MatterportIntegration.tsx

```diff
@@ -0,0 +1,220 @@
+          description: 'Select a connection placeholder',
+        }),
+        value: 'n/a',
+      },
+    ];
+    const get3pConnectionListFunction = getGlobalSettings().get3pConnectionListFunction;
+    if (get3pConnectionListFunction) {
+      const response = await get3pConnectionListFunction('AWSIoTTwinMaker_Matterport');
+      if (response) {
+        response.forEach((secret) => {
+          if (secret.Name && secret.ARN) {
+            connectionList.push({ label: secret.Name, value: secret.ARN });
+          }
+        });
+      }
+    }
+    setConnectionOptions(connectionList);
+  };
+
+  const onUpdated = () => {
+    updateSelectedConnectionName();
+    getConnectionList();
+  };
+
+  useEffect(() => {
+    subscribe(onUpdated);
+    updateSelectedConnectionName();
+    getConnectionList();
+
+    return () => unsubscribe(onUpdated);
+  }, []);
+
+  const onConnectionNameChange = useCallback(async (event) => {
+    const value = event.detail.selectedOption.value;
+    setSelectedConnectionName(value);
+
+    const getSceneInfoFunction = getGlobalSettings().getSceneInfoFunction;
+    const updateSceneFunction = getGlobalSettings().updateSceneFunction;
+    if (getSceneInfoFunction && updateSceneFunction) {
+      let sceneCapabilities: string[] | undefined;
+      let sceneMetadata: Record<string, string> | undefined;
+      const getSceneresponse = await getSceneInfoFunction();
+      if (getSceneresponse) {
+        sceneCapabilities = getSceneresponse.capabilities;
+        sceneMetadata = getSceneresponse.sceneMetadata;
+      }
+
+      if (value === 'n/a') {
+        if (sceneCapabilities && sceneCapabilities.includes('MATTERPORT')) {
+          sceneCapabilities = sceneCapabilities.filter((capability) => capability !== 'MATTERPORT');
+        }
+
+        if (sceneMetadata && sceneMetadata.MATTERPORT_SECRET_ARN) {
+          delete sceneMetadata.MATTERPORT_SECRET_ARN;
+        }
+      } else {
+        if (sceneCapabilities) {
+          if (!sceneCapabilities.includes('MATTERPORT')) {
+            sceneCapabilities.push('MATTERPORT');
+          }
+        } else {
+          sceneCapabilities = ['MATTERPORT'];
+        }
+
+        if (sceneMetadata) {
+          sceneMetadata.MATTERPORT_SECRET_ARN = value;
+        } else {
+          sceneMetadata = { MATTERPORT_SECRET_ARN: value };
+        }
+      }
+
+      updateSceneFunction(sceneCapabilities, sceneMetadata);
+    }
+  }, []);
+
+  const selectedConnection = () => {
+    const selectedOption = connectionOptions.find((t) => t.value === selectedConnectionName);
+    if (selectedOption) {
+      return {
+        label: selectedOption.label,
+        value: selectedOption.value,
+      };
+    }
+    return null;
+  };
+
+  const onMatterportModelIdChange = useCallback(
+    (event) => {
+      const value = event.detail.value;
+      if (value !== matterportModelId) {
+        setMatterportModelId(value);
+        setDirty(true);
```

Comment:
it doesn't seem like `dirty` flag makes any difference to just call `setSceneProperty` here

Reply:
dirty flag prevents calling setSceneProperty during the first loading.